drivers/overlay/overlay.go

```diff
@@ -2060,7 +2083,41 @@ func (d *Driver) DiffGetter(id string) (graphdriver.FileGetCloser, error) {
 	if err != nil {
 		return nil, err
 	}
-	return &overlayFileGetter{diffDirs: append([]string{p}, paths...)}, nil
+
+	// map from diff dir to the directory with the composefs blob mounted
+	composefsMounts := make(map[string]*os.File)
+	defer func() {
+		if Err != nil {
+			for _, f := range composefsMounts {
+				f.Close()
+				unix.Rmdir(f.Name())
+			}
+		}
+	}()
+	diffDirs := append([]string{p}, paths...)
+	for _, diffDir := range diffDirs {
+		// diffDir has the form $GRAPH_ROOT/overlay/$ID/diff, so grab the $ID from the parent directory
+		id := path.Base(path.Dir(diffDir))
+		composefsBlob := d.getComposefsData(id)
+		if fileutils.Exists(composefsBlob) != nil {
```

Comment:
This is only very tangentially related but I find the fact that someone went out of their way to create `fileutils.Exists` but made it return an *error* type instead of a `bool` confusing.  Reading `if fileutils.Exists(composefsBlob) != nil {` I have to stop and think for a second, but `if fileutils.Exists(composefsBlob) {` just would read more nicely...

Of course in Rust, there's https://doc.rust-lang.org/std/path/struct.Path.html#method.try_exists which is *really* what we want here; we should still throw an error IMO if we get e.g. `EIO` or something else.  Anyways.

Reply:
we still need the error in case `Exists` fails, so it would be `exists, err := fileutils.Exists()` and `exists` is true only when `err == nil`

The `fileutils.Exists()` function is a replacement for the `_, err := os.Stat(...)` idiom that becomes `err := fileutils.Exists(...)`, so it is also easier to replace in an existing code base

Reply:
Yeah sorry this is a minor irrelevant tangent, just marking as resolved